Controller/Adminhtml/Menu/Edit.php

```diff
@@ -1,58 +1,100 @@
+     * @param PageFactory $resultPageFactory
+     * @param MenuRepositoryInterface $menuRepository
+     * @param MenuFactory $menuFactory
+     */
+    public function __construct(
+        Context $context,
+        Registry $coreRegistry,
+        PageFactory $resultPageFactory,
+        MenuRepositoryInterface $menuRepository,
+        MenuFactory $menuFactory
+    ) {
+        $this->resultPageFactory = $resultPageFactory;
         $this->menuRepository = $menuRepository;
-        $this->registry = $registry;
+        $this->menuFactory = $menuFactory;
+        $this->coreRegistry = $coreRegistry;
+        parent::__construct(
+            $context
+        );
     }
 
     /**
-     * Dispatch request
-     *
-     * @return \Magento\Framework\Controller\ResultInterface|ResponseInterface
-     * @throws \Magento\Framework\Exception\NotFoundException
+     * @SuppressWarnings(PHPMD.NPathComplexity)
      */
-    public function execute()
+    public function execute(): ResultInterface
     {
-        $id = $this->getRequest()->getParam('id');
-        try {
-            $model = $this->menuRepository->getById($id);
-            $this->registry->register(self::REGISTRY_CODE, $model);
-            $result = $this->resultFactory->create(ResultFactory::TYPE_PAGE);
-            $result->setActiveMenu('Snowdog_Menu::menus');
-            $result->getConfig()->getTitle()->prepend(__('Edit Menu %1', $model->getTitle()));
-            return $result;
-        } catch (NoSuchEntityException $e) {
-            $result = $this->resultRedirectFactory->create();
-            $result->setPath('*/*/index');
-            return $result;
+        $menuId = (int) $this->getRequest()->getParam(self::ID);
+
+        if ($menuId) {
+            try {
+                $model = $this->menuRepository->getById($menuId);
+            } catch (NoSuchEntityException $exception) {
+                $this->messageManager->addErrorMessage(__('This menu no longer exists.'));
+                /** @var Redirect $resultRedirect */
+                $resultRedirect = $this->resultRedirectFactory->create();
+
+                return $resultRedirect->setPath('*/*/index');
+            }
+        } else {
+            $model = $this->menuFactory->create();
         }
-    }
 
-    protected function _isAllowed()
-    {
-        return $this->_authorization->isAllowed('Snowdog_Menu::menus');
+        $this->coreRegistry->register(self::REGISTRY_CODE, $model);
+
+        /** @var Page $resultPage */
+        $resultPage = $this->resultPageFactory->create();
+        $this->initPage($resultPage)->addBreadcrumb(
+            $menuId ? __('Edit Menu %1', $model->getTitle()) : __('New Menu'),
+            $menuId ? __('Edit Menu %1', $model->getTitle()) : __('New Menu')
+        );
+        $resultPage->getConfig()->getTitle()->prepend(__('Menus'));
+        $resultPage->getConfig()->getTitle()->prepend($model->getId() ? $model->getTitle() : __('New Menu'));
```

Comment:
extract `$resultPage->getConfig()->getTitle()` to some variable because you use it twice

Reply:
Done

Reply:
I have tried, but there is no setTitle() method to set it after declaring the getTitle() method as a variable. That is why I left it as it.

Reply:
You're right Samir, I have checked `getTitle()` method and any time is called, it builds the page through layouts preparation, therefore, if assigned to a variable, it'll not prepend the title